projects/RabbitMQ.Client/client/api/ConnectionFactory.cs

```diff
@@ -188,6 +189,16 @@ public sealed class ConnectionFactory : ConnectionFactoryBase, IAsyncConnectionF
 
         // just here to hold the value that was set through the setter
         private Uri _uri;
+        private ArrayPool<byte> _pool;
+
+        /// <summary>
+        /// The memory pool used for allocating buffers. Default is <see cref="MemoryPool{T}.Shared"/>.
+        /// </summary>
+        public ArrayPool<byte> MemoryPool
+        {
+            get => _pool ?? ArrayPool<byte>.Shared;
+            set => _pool = value;
+        }
```

Comment:
```suggestion
        private ArrayPool<byte> _pool = ArrayPool<byte>.Shared;

        /// <summary>
        /// The memory pool used for allocating buffers. Default is <see cref="MemoryPool{T}.Shared"/>.
        /// </summary>
        public ArrayPool<byte> MemoryPool
        {
            get => _pool;
            set => _pool = value;
        }
```

Reply:
Due to absence of nullability analysis and annotations on old C# we need to check `value` on **null** and throw exception.

Reply:
Fixed